src/main/java/nl/opengeogroep/safetymaps/server/stripes/SafetyConnectProxyActionBean.java

```diff
@@ -70,56 +83,143 @@ public Resolution proxy() throws Exception {
-            final MutableObject<String> contentType = new MutableObject<>("text/plain");
-            final String content = client.execute(req, new ResponseHandler<String>() {
-                @Override
-                public String handleResponse(HttpResponse hr) {
-                    log.debug("proxy for user " + context.getRequest().getRemoteUser() + " URL " + req.getURI() + ", response: " + hr.getStatusLine().getStatusCode() + " " + hr.getStatusLine().getReasonPhrase());
-                    contentType.setValue(hr.getEntity().getContentType().getValue());
-                    try {
-                        return IOUtils.toString(hr.getEntity().getContent(), "UTF-8");
-                    } catch(IOException e) {
-                        log.error("Exception reading HTTP content", e);
-                        return "Exception " + e.getClass() + ": " + e.getMessage();
+        try(Connection c = DB.getConnection()) {
+            String qs = context.getRequest().getQueryString();
+            final HttpUriRequest req = RequestBuilder.get()
+                    .setUri(url + "/" + path + (qs == null ? "" : "?" + qs))
+                    .addHeader("Authorization", authorization)
+                    .build();
+
+            try(CloseableHttpClient client = HttpClients.createDefault()) {
+                final MutableObject<String> contentType = new MutableObject<>("text/plain");
+                final String content = client.execute(req, new ResponseHandler<String>() {
+                    @Override
+                    public String handleResponse(HttpResponse hr) {
+                        log.debug("proxy for user " + context.getRequest().getRemoteUser() + " URL " + req.getURI() + ", response: " + hr.getStatusLine().getStatusCode() + " " + hr.getStatusLine().getReasonPhrase());
+                        contentType.setValue(hr.getEntity().getContentType().getValue());
+                        try {
+                            return IOUtils.toString(hr.getEntity().getContent(), "UTF-8");
+                        } catch(IOException e) {
+                            log.error("Exception reading HTTP content", e);
+                            return "Exception " + e.getClass() + ": " + e.getMessage();
+                        }
                     }
+                });
+                
+                final String authorizedContent;
+                if (path.toLowerCase().startsWith("incident") && qs != null && qs.length() > 0) {
+                    authorizedContent = applyAuthorizationToIncidentContent(content, c);
+                } else if (path.toLowerCase().startsWith("incident") && (qs == null || qs.length() == 0)) {
+                    throw new Exception("Incident request incomplete.");
+                } else if(path.toLowerCase().startsWith("eenheidlocatie")) {
+                    authorizedContent = applyAuthorizationToEenheidLocatieContent(content);
+                } else {
```

Comment:
This means all other paths are always passed through, but this makes it easy to forget applying authorizations for new API endpoints. The secure default would be to only pass-through unmodified responses for known paths that do not contain user-dependant authorizations.
And does `/eenheidstatus` need to be handled for users which do not have `ROLE_EIGEN_VOERTUIGNUMMER`? The status of other units may not be that sensitive, but if that determination has been made at least add a comment stating that.